### PR TITLE
Overlay appears only on marketplace items

### PR DESCRIPTION
**Pull request by davidkelly**
Which really means entities you know are marketplace items.  Which right now, means things _you_ got from the marketplace.

### CODE REVIEW

File: interface/src/ui/overlays/ContextOverlayInterface.cpp [hunk 1]
```diff
@@ -35,27 +36,30 @@ ContextOverlayInterface::ContextOverlayInterface() {
 void ContextOverlayInterface::createOrDestroyContextOverlay(const EntityItemID& entityItemID, const PointerEvent& event) {
     if (event.getButton() == PointerEvent::SecondaryButton) {
         qCDebug(context_overlay) << "Creating Context Overlay on top of entity with ID: " << entityItemID;
```
Comment (zfox23) [inline interface/src/ui/overlays/ContextOverlayInterface.cpp]: Please move this debug line underneath the next `if()`, so we only get this if we're actually creating the overlay :)

File: interface/src/ui/overlays/ContextOverlayInterface.h [hunk 2]
```diff
@@ -56,6 +56,8 @@ public slots:
 private:
     bool _verboseLogging { true };
     QUuid _currentEntityWithContextOverlay{};
+    QString _marketplaceID;
```
Comment (zfox23) [inline interface/src/ui/overlays/ContextOverlayInterface.h]: Please reset this to empty when we destroy the overlay (inside `ContextOverlayInterface::destroyContextOverlay()`)

Comment (zfox23) [inline interface/src/ui/overlays/ContextOverlayInterface.h]: Also, a little nitpicky, but perhaps name this to something like `_contextualMarketplaceID` to make it easier to search for (and clearer that it isn't some generic Marketplace ID)?

Comment (davidkelly) [inline interface/src/ui/overlays/ContextOverlayInterface.h]: ohhh yes -- doh!

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1781/HighFidelity-Beta-PR10998-920ac17caf0d7b002c5548995e15cb5b2fc9c1f0.exe

**Pull request merged**